editor/src/components/canvas/canvas-strategies/relative-move-strategy.tsx

```diff
@@ -0,0 +1,188 @@
+
+  name: () => 'Move (Relative)',
+
+  isApplicable: (canvasState, _interactionState, metadata) => {
+    const selectedElements = getTargetPathsFromInteractionTarget(canvasState.interactionTarget)
+    if (selectedElements.length === 0) {
+      return false
+    }
+
+    const filteredSelectedElements = getDragTargets(selectedElements)
+    const last = filteredSelectedElements[filteredSelectedElements.length - 1]
+    const meta = MetadataUtils.findElementByElementPath(metadata, last)
+    if (!meta) {
+      return false
+    }
+
+    // should we also support absolute elements, for which we'll do something like `CONVERT_TO_RELATIVE`?
+    return meta.specialSizeMeasurements.position === 'relative'
+  },
+
+  controlsToRender: [],
+
+  fitness: (canvasState, interactionState, _sessionState) => {
+    const { interactionData, activeControl } = interactionState
+    if (!(interactionData.type === 'DRAG' && interactionData.drag != null)) {
+      return 0
+    }
+    if (activeControl.type !== 'BOUNDING_AREA') {
+      return 0
+    }
+
+    const selectedElements = getTargetPathsFromInteractionTarget(canvasState.interactionTarget)
+    if (selectedElements.length === 0) {
+      return 0
+    }
+    const filteredSelectedElements = getDragTargets(selectedElements)
+    const last = filteredSelectedElements[filteredSelectedElements.length - 1]
+    const meta = MetadataUtils.findElementByElementPath(interactionState.latestMetadata, last)
+    if (!meta) {
+      return 0
+    }
+    const offsets = getStyleOffsets(meta)
+    if (!offsets) {
+      return 0
+    }
+
+    const hasOffsets =
+      offsets.left != null || offsets.top != null || offsets.right != null || offsets.bottom != null
+
+    return hasOffsets
+      ? 4 // +1 than reorder flow
+      : 1 // there should be a more structured way to define priorities (:
+  },
+
+  apply: (canvasState, interactionState, sessionState) => {
+    const { interactionData } = interactionState
+    if (!(interactionData.type === 'DRAG' && interactionData.drag != null)) {
+      return emptyStrategyApplicationResult
+    }
+
+    const selectedElements = getTargetPathsFromInteractionTarget(canvasState.interactionTarget)
+    if (selectedElements.length === 0) {
+      return emptyStrategyApplicationResult
+    }
+
+    const filteredSelectedElements = getDragTargets(selectedElements)
+    const last = filteredSelectedElements[filteredSelectedElements.length - 1]
+    const meta = MetadataUtils.findElementByElementPath(sessionState.startingMetadata, last)
+    if (meta == null) {
+      return emptyStrategyApplicationResult
+    }
+    const offsets = getStyleOffsets(meta)
+    if (!offsets) {
+      return emptyStrategyApplicationResult
+    }
+
+    const { drag } = interactionData
+
+    const commands: Array<CanvasCommand> = []
+
+    // should set a direction only if it's either defined in the original offsets or if the counterpart (horiz/vert) is not defined
+    // vertical
+    commands.push(
```

Comment:
I think this should use/share code with `applyAbsoluteMoveCommon` (and probably the Absolute should be removed from the name of the shared function :)) because that function also provides some common drag-to-move behavior such as snapping, holding shift locks the move to one axis, and holding cmd disables snapping.

Reply:
Yeah this is a good point. We don't want to lose / overlook that common functionality here.

Reply:
something like this? https://github.com/concrete-utopia/utopia/pull/2604/commits/1c4e6b29855e726829728377778956dd55d19261

Reply:
yep, perfect!